suite-native/module-connect-popup/src/hooks/useConnectPopupNavigation.ts

```diff
@@ -34,33 +34,12 @@ export const useConnectPopupNavigation = () => {
     const featureFlagEnabled = useFeatureFlag(FeatureFlag.IsConnectPopupEnabled);
     const navigation = useNavigation<NavigationProp>();
 
-    const navigateToConnectPopup = useCallback(
-        (url: string) => {
-            if (!featureFlagEnabled) return;
-            if (!isConnectPopupUrl(url)) return;
-            const parsedUrl = Linking.parse(url);
-            navigation.navigate(RootStackRoutes.ConnectPopup, { parsedUrl });
-        },
-        [navigation, featureFlagEnabled],
-    );
+    const url = Linking.useURL();
 
     useEffect(() => {
-        const navigateToInitalUrl = async () => {
-            const currentUrl = await Linking.getInitialURL();
-            if (currentUrl) {
-                navigateToConnectPopup(currentUrl);
-            }
-        };
-        navigateToInitalUrl();
-    }, [navigateToConnectPopup]);
-
-    useEffect(() => {
-        // there could be when you open same deep link for second time and in that case it will be ignored
-        // this could be probably handed by Linking.addEventListener
-        const subscription = Linking.addEventListener('url', event => {
-            navigateToConnectPopup(event.url);
-        });
-
-        return () => subscription?.remove();
-    }, [navigateToConnectPopup]);
+        if (!featureFlagEnabled) return;
+        if (!url || !isConnectPopupUrl(url)) return;
+        const parsedUrl = Linking.parse(url);
```

Comment:
I would suggest wrapping parse to try catch

Reply:
Good point, done